contracts/global-extensions/GlobalAuctionRebalanceExtension.sol

```diff
@@ -0,0 +1,266 @@
+/*
+    Copyright 2023 Index Coop
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import { AddressArrayUtils } from "../lib/AddressArrayUtils.sol";
+import { BaseGlobalExtension } from "../lib/BaseGlobalExtension.sol";
+import { IAuctionRebalanceModuleV1 } from "../interfaces/IAuctionRebalanceModuleV1.sol";
+import { IManagerCore } from "../interfaces/IManagerCore.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+
+
+/**
+ * @title GlobalAuctionRebalanceExtension
+ * @author Index Coop
+ *
+ * @dev Extension contract for interacting with the AuctionRebalanceModuleV1. This contract acts as a pass-through and functions 
+ * are only callable by operator. 
+ */
+contract GlobalAuctionRebalanceExtension is BaseGlobalExtension {
+    using AddressArrayUtils for address[];
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event AuctionRebalanceExtensionInitialized(
+        address indexed _setToken,
+        address indexed _delegatedManager
+    );
+
+
+    /* ============ Structs ============ */
+
+    struct AuctionExecutionParams {
+        uint256 targetUnit;                      // Target quantity of the component in Set, in precise units (10 ** 18).
+        string priceAdapterName;                 // Identifier for the price adapter to be used.
+        bytes priceAdapterConfigData;            // Encoded data for configuring the chosen price adapter.
+    }
+
+    /* ============ State Variables ============ */
+    
+    IAuctionRebalanceModuleV1 public immutable auctionModule;  // AuctionRebalanceModuleV1
+
+
+    /* ============ Constructor ============ */
+
+    constructor(IManagerCore _managerCore, IAuctionRebalanceModuleV1 _auctionModule) public BaseGlobalExtension(_managerCore) {
```

Comment:
nit: formatting and natspec to match 
https://github.com/IndexCoop/index-coop-smart-contracts/blob/936ccf83bfd75e7b52b4df231bd8356216d6f6ac/contracts/global-extensions/GlobalWrapExtension.sol#L55-L69